app/javascript/tests/close_contact/CloseContact.test.js

```diff
@@ -0,0 +1,219 @@
+    expect(alreadyExistingCCWrapper.state('email')).toEqual(mockCloseContact2.email || '')
+    expect(alreadyExistingCCWrapper.state('last_date_of_exposure')).toEqual(mockCloseContact2.last_date_of_exposure || null)
+    expect(alreadyExistingCCWrapper.state('assigned_user')).toEqual(mockCloseContact2.assigned_user || null)
+    expect(alreadyExistingCCWrapper.state('notes')).toEqual(mockCloseContact2.notes || '')
+    expect(alreadyExistingCCWrapper.state('contact_attempts')).toEqual(mockCloseContact2.contact_attempts || 0)
+  });
+
+  it('Properly renders the modal if button is clicked', () => {
+    const emptyCCWrapper = getShallowWrapper(mockCloseContact1);
+    expect(emptyCCWrapper.state('showModal')).toBeFalsy();
+    emptyCCWrapper.find(Button).simulate('click');
+    expect(emptyCCWrapper.state('showModal')).toBeTruthy();
+    expect(emptyCCWrapper.find(Modal.Header).exists()).toBeTruthy();
+    expect(emptyCCWrapper.find(Modal.Header).find('.sr-only').text()).toEqual('Close Contact');
+    expect(emptyCCWrapper.find(Modal.Header).find('ModalTitle').text()).toEqual('Close Contact');
+    expect(emptyCCWrapper.find(Modal.Body).exists()).toBeTruthy();
+    // Using `toContain` instead of `toEqual` to avoid any whitespace issues
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(0).find('FormLabel').text()).toContain(`First Name *`);
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(1).find('FormLabel').text()).toContain(`Last Name *`);
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(2).find('FormLabel').text()).toContain(`Phone Number`);
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(3).find('FormLabel').text()).toContain(`Email`);
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(4).find('FormLabel').text()).toContain(`Last Date of Exposure`);
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(5).find('FormLabel').text()).toContain(`Assigned User`); // There's also the InfoTooltip
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(5).find('FormLabel').containsMatchingElement(<InfoTooltip />)).toBeTruthy();
+    expect(emptyCCWrapper.find(InfoTooltip).prop('tooltipTextKey')).toEqual('assignedUser');
+
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(6).find('FormLabel').at(0).text()).toContain('Notes');
+    expect(emptyCCWrapper.find(Modal.Body).find('Row').at(6).find('FormLabel').at(1).text()).toContain('2000 characters remaining');
+    expect(emptyCCWrapper.find(Modal.Footer).exists()).toBeTruthy();
+    expect(emptyCCWrapper.find(Modal.Footer).find(Button).at(0).text()).toEqual('Cancel');
+    expect(emptyCCWrapper.find(Modal.Footer).find(Button).at(1).text()).toEqual('Create');
+    expect(emptyCCWrapper.find(Button).at(2).props().disabled).toBeTruthy()
+  });
+
+  it('Properly renders empty fields when an empty close contact is used as a prop', () => {
```

Comment:
The name of this test makes it sound like it's the same as the first one above:
> Properly renders all main components for empty close contact

But in reality it seems to be checking that the onChange handler is called for each input. I would suggest renaming.

Reply:
Renamed in latest commit